### PR TITLE
PR to get Nancy on DNX

### PR DESCRIPTION
**Pull request by jchannon**

### CODE REVIEW

**Comment by khellang** [comment]
It looks like this PR contains commits that shouldn't be there? Specifically e6886b69d350e216642d6ac892f481b4cfdbea25, 857a00268a13cd39713e60e1dcf9f03033c3fe39, 7d53b3b592227036a10fc7c933eef158eddcfc49 and 7bc6bebb124293491cad08beb449514ad17332c3.

**Comment by jchannon** [comment]
I dont know whats happened. Feel free to submit your own PR, I can't be fucked

File: src/Nancy.Authentication.Basic.Tests/project.json [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{
+    "version": "1.0.0-*",
+    "description": "",
+    "authors": [ "" ],
+    "tags": [ "" ],
+    "projectUrl": "",
+    "licenseUrl": "",
+
+    "dependencies": {
+        "Nancy": "1.0.0-*",
```
Comment (khellang) [inline src/Nancy.Authentication.Basic.Tests/project.json]: Make these `"Nancy": { "target": "project" }`

File: src/Nancy.Authentication.Basic.Tests/project.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+    "version": "1.0.0-*",
```
Comment (khellang) [inline src/Nancy.Authentication.Basic.Tests/project.json]: `2.0.0-*`?

File: src/Nancy.Authentication.Basic.Tests/project.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{
+    "version": "1.0.0-*",
+    "description": "",
+    "authors": [ "" ],
```
Comment (khellang) [inline src/Nancy.Authentication.Basic.Tests/project.json]: All of these should be filled out :smile:

File: src/Nancy.Authentication.Basic.Tests/project.json [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    "dependencies": {
+        "Nancy": "1.0.0-*",
+        "FakeItEasy": "1.19.0",
+        "xunit": "1.9.1",
+        "xunit.extensions": "1.9.1",
+        "Nancy.Authentication.Basic": "1.0.0-*"
+
+    },
+    "compileFiles": [ "../Nancy.Tests/ShouldExtensions.cs", "../Nancy.Tests/Fakes/FakeRequest.cs", "../SharedAssemblyInfo.cs" ],
```
Comment (khellang) [inline src/Nancy.Authentication.Basic.Tests/project.json]: We should kill the `SharedAssemblyInfo` at this point, no?

File: src/Nancy.Demo.Authentication.Stateless.Website/Web.config [hunk 5]
```diff
@@ -1,7 +1,19 @@
 ﻿<?xml version="1.0"?>
-
+<!--
+  For more information on how to configure your ASP.NET application, please visit
+  http://go.microsoft.com/fwlink/?LinkId=169433
+  -->
 <configuration>
+  <!--
```
Comment (khellang) [inline src/Nancy.Demo.Authentication.Stateless.Website/Web.config]: I'm guessing these comments came from the .NET 4.5 update commits that shouldn't be here? Otherwise we should kill these.

File: src/Nancy.Demo.Authentication.Stateless/UserDatabase.cs [hunk 6]
```diff
@@ -6,6 +6,8 @@ namespace Nancy.Demo.Authentication.Stateless
     using System.Security.Claims;
     using System.Security.Principal;
 
+    using Nancy.Security;
```
Comment (khellang) [inline src/Nancy.Demo.Authentication.Stateless/UserDatabase.cs]: Why was this added?

File: src/Nancy.Authentication.Basic/project.json [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+  "licenseUrl": "",
+
+  "dependencies": {
+    "Nancy": "1.0.0-*"
+
+  },
+
+  "frameworks": {
+    "dnx451": {
+      "frameworkAssemblies": { "System.Xml": "" }
```
Comment (khellang) [inline src/Nancy.Authentication.Basic/project.json]: Version?

File: src/Nancy.Demo.Authentication.Forms.TestingDemo/project.json [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+    "xunit.extensions": "1.9.1",
+    "xunit": "1.9.1",
+    "Nancy.Testing": "1.0.0-*",
+    "Nancy.Demo.Authentication.Forms": "1.0.0-*"
+
+  },
+  "compileFiles": [ "../Nancy.Tests/ShouldExtensions.cs", "../SharedAssemblyInfo.cs" ],
+  "frameworks": {
+    "dnx451": {
+      "frameworkAssemblies": { }
```
Comment (khellang) [inline src/Nancy.Demo.Authentication.Forms.TestingDemo/project.json]: :first:

File: src/Nancy.Demo.Authentication.Forms.TestingDemo/project.json [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+  "projectUrl": "",
+  "licenseUrl": "",
+
+  "dependencies": {
+    "Nancy": "1.0.0-*",
+    "xunit.extensions": "1.9.1",
+    "xunit": "1.9.1",
+    "Nancy.Testing": "1.0.0-*",
+    "Nancy.Demo.Authentication.Forms": "1.0.0-*"
+
```
Comment (khellang) [inline src/Nancy.Demo.Authentication.Forms.TestingDemo/project.json]: :fire: (BTW, most of these comments go for *all* project.json

File: src/Nancy.Demo.CustomModule/project.json [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+  "tags": [ "" ],
+  "projectUrl": "",
+  "licenseUrl": "",
+
+  "dependencies": {
+    "Nancy": "1.0.0-*",
+
+
+  },
+  "compileFiles": [  ],
```
Comment (khellang) [inline src/Nancy.Demo.CustomModule/project.json]: Loads of copy/pasta here... :disappointed:

File: src/Nancy.Demo.Hosting.Dnx/project.json [hunk 11]
```diff
@@ -0,0 +1,35 @@
+﻿{
+    "webroot": "wwwroot",
+    "version": "1.0.0-*",
+
+    "dependencies": {
+        "Nancy": "1.0.0-*",
+        "Microsoft.AspNet.Owin":  "1.0.0-beta5",
```
Comment (khellang) [inline src/Nancy.Demo.Hosting.Dnx/project.json]: Update to `-rc1-final`?

File: src/Nancy.Demo.Hosting.Self/project.json [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    "frameworks": {
+        "dnx451": {
+            "frameworkAssemblies": {
+                "System.Diagnostics.Debug": ""
+            }
+        },
+        "dnxcore50": {
+            "dependencies": {
+                "System.Console": "4.0.0-beta-23029",
```
Comment (khellang) [inline src/Nancy.Demo.Hosting.Self/project.json]: `-beta-23516`

File: src/Nancy.Hosting.Self/FileSystemRootPathProvider.cs [hunk 13]
```diff
@@ -1,17 +1,34 @@
...
+#endif
 
     public class FileSystemRootPathProvider : IRootPathProvider
     {
+        private readonly IApplicationEnvironment env;
+
+        public FileSystemRootPathProvider()
+        {
+#if DNX451
+            this.env = CallContextServiceLocator.Locator.ServiceProvider.GetService(typeof(IApplicationEnvironment)) as IApplicationEnvironment;
```
Comment (khellang) [inline src/Nancy.Hosting.Self/FileSystemRootPathProvider.cs]: I'd rather see this factored out into a separate service so it could be mocked and tested.

File: src/Nancy.Hosting.Self/project.json [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    "Microsoft.Framework.Runtime.Abstractions": "1.0.0-beta5-*"
+  },
+  "compileFiles": [ "../SharedAssemblyInfo.cs" ],
+  "frameworks": {
+    "dnx451": {
+      "frameworkAssemblies": {
+        "System.Diagnostics.Debug": ""
+      }
+    },
+    "dnxcore50": {
```
Comment (khellang) [inline src/Nancy.Hosting.Self/project.json]: Maybe we should kill these until we have a better story on .NET core? Seems like this PR is biting off a bit more than it can chew.

File: src/Nancy.Hosting.Wcf.Tests/project.json [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+  "version": "1.0.0-*",
+  "description": "",
+  "authors": [ "" ],
+  "tags": [ "" ],
+  "projectUrl": "",
+  "licenseUrl": "",
+
+  "dependencies": {
+    "Nancy": "1.0.0-*",
+    "Nancy.Hosting.Wcf": "1.0.0-*",
```
Comment (khellang) [inline src/Nancy.Hosting.Wcf.Tests/project.json]: Nancy.Hosting.Wcf is dead? How can this compile?

File: src/Nancy.Testing/TestingViewContextKeys.cs [hunk 16]
```diff
@@ -1,25 +0,0 @@
-﻿namespace Nancy.Testing
-{
-    /// <summary>
-    /// The key names for where the testing view context data is stored
-    /// </summary>
-    public static class TestingViewContextKeys
```
Comment (khellang) [inline src/Nancy.Testing/TestingViewContextKeys.cs]: Was this class unused?
